### PR TITLE
Add routing params to request query params for middleware usage

### PR DESCRIPTION
**Pull request by delboy1978uk**
As per issue here https://github.com/thephpleague/route/issues/249

### CODE REVIEW

**Comment by odan** [comment]
From my point of view: Routing args and a query string is conceptionally someting different and should not be mixed.

Mixing the query string arguments (`?id=1`) and the route vars (`/users/{id:[0-9]+}`) could cause some naming **conflicts** and could break working code.

**Comment by delboy1978uk** [comment]
A `routing var` as you put it, is just a `pretty URL`. So in most folks eyes I think they would agree that this is indeed a query parameter. 
The only example (using your example) I can think would break the code would be
`/users/5?id=6`
But I doubt anyone would do that.

**Comment by delboy1978uk** [comment]
I forgot to tag this issue in the pull request https://github.com/thephpleague/route/issues/249. You can't get any routing params in any middleware, so we do need this!

**Comment by delboy1978uk** [comment]
Just had a discussion in ##php and it has been suggested than the route vars get put into PSR-7 ServerRequestInterface attributes. https://github.com/php-fig/http-message/blob/master/src/ServerRequestInterface.php#L202:L204 The docblock itself says:
The request "attributes" may be used to allow injection of any parameters derived from the request: e.g. the results of path match operations, and that sounds exactly like what we need. I'll work on refactoring the array merge out and injecting route vars as attributes instead. Thanks for your input @odan :-)

**Comment by philipobenito** [comment]
sorry for the delay on this, looking at a release the first week in the new year so will look at bringing this in for that

**Comment by jan-di** [comment]
Hello @philipobenito, is there anything how we can help to get this pull request merged? I would like to use route specific vars in a Middleware and i think this is the one big missing part of this router, to make it completly middleware ready 🙂 
Thanks

**Comment by philipobenito** [comment]
Ah, I got delayed on the major release and forgot about this, will take a look tomorrow

Sent from ProtonMail mobile

-------- Original Message --------

File: src/Dispatcher.php [hunk 1]
```diff
@@ -42,12 +42,29 @@ public function dispatchRequest(ServerRequestInterface $request): ResponseInterf
...
 
         return $this->handle($request);
     }
 
+    /**
+     * @param ServerRequestInterface $request
+     * @param Route $route
+     * @return ServerRequestInterface
+     */
+    private function requestWithRouteAttributes(ServerRequestInterface $request, Route $route): ServerRequestInterface
```
Comment (philipobenito) [inline src/Dispatcher.php]: `protected` please

File: src/Dispatcher.php [hunk 2]
```diff
@@ -42,12 +42,29 @@ public function dispatchRequest(ServerRequestInterface $request): ResponseInterf
...
                 $this->setFoundMiddleware($route);
+                $request = $this->requestWithRouteAttributes($request, $route);
                 break;
         }
 
         return $this->handle($request);
     }
 
+    /**
+     * @param ServerRequestInterface $request
```
Comment (philipobenito) [inline src/Dispatcher.php]: Please format doc comments in line with others

File: tests/DispatchIntegrationTest.php [hunk 3]
```diff
@@ -19,7 +19,7 @@ class DispatchIntegrationTest extends TestCase
      *
      * @return void
      */
-    public function testDispatchesFoundRoute(): void
+ /*   public function testDispatchesFoundRoute(): void
```
[the remark this answers is not in the document]
Comment (delboy1978uk) [reply tests/DispatchIntegrationTest.php]: oh! let me sort that just now!! apologies

**Comment by philipobenito** [comment]
I'll look at getting these merged and released at some point today or tomorrow 👍

**Comment by philipobenito** [comment]
I’ll get to these next week, had some personal stuff going on so got myself behind on things

**Pull request merged**